src/time.rs

```diff
@@ -0,0 +1,117 @@
+use std::{
+    ops::Deref,
+    time::{Duration, Instant},
+};
+
+use bevy::prelude::*;
+
+use crate::{
+    AdvanceWorld, AdvanceWorldSet, CloneStrategy, ResourceSnapshotPlugin, RollbackFrameCount,
+    DEFAULT_FPS,
+};
+
+/// [`Resource`] describing the rate at which the [`AdvanceWorld`] will run.
+///
+/// Updating this value after the session has started may produce
+/// unintended results and should be avoided.
+#[derive(Resource, Clone, Copy, Debug, Hash)]
+pub struct RollbackFrameRate(pub usize);
+
+impl Default for RollbackFrameRate {
+    fn default() -> Self {
+        Self(DEFAULT_FPS)
+    }
+}
+
+/// A [`Resource`] which manages a [`Time`] directly tied to the [`GgrsFrameRate`].
+///
+/// This resource is available at all times, and will be updated during [`AdvanceWorld`].
+/// During this schedule, the Bevy managed [`Time`] will be replaced with one from
+/// this resource.
+#[derive(Resource, Clone)]
+pub struct RollbackTime {
```

Comment:
Should this be `pub(crate)`? Is there a reason for the user to interact with this instead of `Time`?

Reply:
I left this public just in case an end-user wanted to get access to the game-time from outside the rollback schedules. It could be private to the crate to reduce the API surface tho.

Reply:
I think it depends on whether we encourage the user to use `Time` or not (is it a mistake or the intended way?).

As it reads now, box_game uses `Time`, so I think it would be best to hide as an implementation detail.